requirements-dev.txt

```diff
@@ -8,5 +8,4 @@ pytest==3.0.3
 pytest-cov==2.4.0
 pytest-flake8==0.8.1
 pytest-timeout==1.2.0
-requests==2.10.0
```

Comment:
is this removal related to this PR?

Reply:
Actually (and strangely) yes it is.

The `requests` dependency does not seem to be needed for pycsw - I've done some grepp'ing in the codebase and I don't think we are using it anywhere (I might be wrong though).
 
In addition to that (and hence why it is related here), the gunicorn wsgi server, which is being used in the docker image, depends on a more recent version of requests. 
When I was testing the workflow for getting a development environment based on docker (as described in the docs on the other PR) I realized that when one would do 

```
pip install -r requirements-dev.txt
```

inside a running container it would uninstall the version of requests required by gunicorn and downgrade it to the one pinned in our requirements-dev.txt. This would lead to a crash in gunicorn when reloading/restarting the server's worker processes.

Reply:
thanks!